### ISSUE TITLE
Windows: CreateDirectory fails due to empty data_dir

### ISSUE THREAD

**Issue by robhshu** [dbartolini/crown#76]
I've downloaded and built crown on Windows and immediately hit this assert:

> 
> Assertion failed: err != 0
>     In: ..\..\..\src\core\os.cpp:178
>     CreateDirectory: GetLastError = 3
> Stacktrace:
>     [ 1] crown::error::abort in c:\users\rhs\github\crown\src\core\error\error.cpp:26
>     [ 2] crown::error::abort in c:\users\rhs\github\crown\src\core\error\error.cpp:35
>     [ 3] crown::os::create_directory in c:\users\rhs\github\crown\src\core\os.cpp:178
>     [ 4] crown::FilesystemDisk::create_directory in c:\users\rhs\github\crown\src\core\filesystem\filesystem_disk.cpp:326
>     [ 5] crown::Device::run in c:\users\rhs\github\crown\src\device\device.cpp:359
>     [ 6] crown::run in c:\users\rhs\github\crown\src\device\device.cpp:760
>     [ 7] crown::func in c:\users\rhs\github\crown\src\device\main_windows.cpp:337
>     [ 8] crown::Thread::run in c:\users\rhs\github\crown\src\core\thread\thread.cpp:105
>     [ 9] crown::thread_proc in c:\users\rhs\github\crown\src\core\thread\thread.cpp:21

What seems to be happening is `Device::run` is looking for the data directory, and as it isn't set is returning `""` instead of `nullptr` - so instead of falling back on `os::getcwd` to get the current directory - it uses an empty string, which goes through this logic:
```cpp
	const char* data_dir = _device_options._data_dir.c_str(); // returns "", not nullptr
	if (!data_dir) // doesn't hit
	{
		char buf[1024];
		data_dir = os::getcwd(buf, sizeof(buf));
	}
	_data_filesystem = CE_NEW(_allocator, FilesystemDisk)(default_allocator());
	((FilesystemDisk*)_data_filesystem)->set_prefix(data_dir);
	if (!_data_filesystem->exists(data_dir)) // "" does not exist, so this condition is met
		_data_filesystem->create_directory(data_dir); // attempts to create a new directory with ""
```

One fix could be to check the length:

```cpp
_device_options._data_dir.length() > 0
```

**Comment by dbartolini** [comment]
Can you paste the full command line to reproduce this?

**Comment by robhshu** [comment]
I've used the AppVeyor command to generate VS2017 files using this:

```3rdparty\bx\tools\bin\windows\genie --file=scripts\genie.lua --with-luajit --with-openal --with-bullet --with-tools --no-level-editor vs2017```

Then:

* Opened crown **Project Properties**
* Set the **Command Arguments** to `../../../samples/01-physics`
* Hit **Start Local Debugger**

**Comment by dbartolini** [comment]
Should be fixed in: e047eeb4c2b65fb3ea112053aa692f233bb1cb17
To try examples you should launch the engine with:

    > --source-dir C:\crown\samples\02-animation --map-source-dir core C:\crown\samples --compile --continue

**Comment by robhshu** [comment]
Thanks, this gets further but now fails to start `texturec-debug.exe`.

I'll open a sepearate issue when I have more info!
